src/components/Router.js

```diff
@@ -6,22 +6,33 @@ import AddItems from './AddItems';
 import NoMatch from './NoMatch';
 import NewList from './NewList';
 import Welcome from './Welcome';
+import Nav from './Nav';
 
 const Router = () => {
   const currentList = useContext(ListContext);
 
+  // If there's no user token, Add Items and View List redirects to Welcome
+
   return (
-    <Switch>
-      <Route exact path="/">
-        {currentList.token ? <Redirect to="/view-list" /> : <Welcome />}
-      </Route>
-      <Route exact path="/view-list" component={() => <ViewList />} />
-      <Route exact path="/add-items" component={() => <AddItems />} />
-      <Route exact path="/new-list">
-        {currentList.token ? <Redirect to="/view-list" /> : <NewList />}
-      </Route>
-      <Route component={NoMatch} />
-    </Switch>
+    <div>
+      <Switch>
+        <Route exact path="/">
```

Comment:
I think I would avoid the code repetition in here, and abstract this to a PrivateRoute component. Similar to the PrivateRoute in here: https://medium.com/@thanhbinh.tran93/private-route-public-route-and-restricted-route-with-react-router-d50b27c15f5e 

You can easily pass in two components to render if the user  has a token or not. 

Other than that the PR looks great! Good job!

Reply:
Hi Indira, thanks so much for your feedback! We've updated our Router to use private/public routes, and moved the conditional rendering of Nav into PrivateRoute.